catalog/src/main/kotlin/com/adevinta/spark/catalog/icons/IconsScreen.kt

```diff
@@ -0,0 +1,185 @@
+                )
+            },
+        )
+        VerticalSpacer(space = 16.dp)
+        LazyVerticalGrid(
+            modifier = modifier
+                .consumeWindowInsets(contentPadding)
+                .fillMaxSize()
+                .clickable(
+                    // no ripple effect is needed as this onClick is just to clear the focus of the search field
+                    indication = null,
+                    interactionSource = remember { MutableInteractionSource() },
+                ) {
+                    focusManager.clearFocus()
+                },
+
+            contentPadding = contentPadding,
+            columns = GridCells.Adaptive(minSize = 60.dp),
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+            horizontalArrangement = Arrangement.spacedBy(16.dp),
+        ) {
+            items(filteredIcons.size) { index ->
+                val iconWithNamePair = filteredIcons[index]
+                Column(
+                    modifier = Modifier
+                        .clip(SparkTheme.shapes.small)
+                        .combinedClickable(
+                            onLongClick = { copyToClipboard(context, iconWithNamePair.second) },
+                            onClick = {
+                                val route = "$IconDemoRoute/${iconWithNamePair.first}/${iconWithNamePair.second}"
+                                navController.navigate(
+                                    route = route,
+                                )
+                            },
+                        )
+                        .padding(8.dp),
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    verticalArrangement = Arrangement.spacedBy(4.dp),
+                ) {
+                    Icon(
+                        sparkIcon = SparkIcon.DrawableRes(iconWithNamePair.first),
+                        contentDescription = null,
+                        size = IconSize.Large,
+                    )
+                    Text(
+                        text = iconWithNamePair.second,
+                        style = SparkTheme.typography.caption,
+                    )
+                }
+            }
+        }
+    }
+}
+
+/**
+ * @return a [List] of [Pair]s of drawable res Int and String, representing formatted name of the resource
+ */
+private fun getAllIconsRes(context: Context) = IconR.drawable::class.java.declaredFields.map {
+    val icon = it.getInt(null)
+    icon to context.resources.getResourceEntryName(icon).removePrefix("spark_icons_").toPascalCase()
+}
+
+private fun String.toPascalCase(): String = split("_").joinToString(separator = "") { str ->
+    str.replaceFirstChar {
+        if (it.isLowerCase()) {
+            it.titlecase(
+                Locale.ROOT,
+            )
+        } else {
+            it.toString()
+        }
+    }
+}
+
+private fun copyToClipboard(context: Context, text: String) {
+    val clipboardManager =
```

Comment:
Don't we have access to the extension `getSystemService<ClipboardManager>()`

Reply:
Yes, we actually do! Thanks!